### ISSUE TITLE
[Build]: The amount of dependencies is spiraling out of control. 

### ISSUE THREAD

**Issue by gir489returns** [YimMenu/YimMenu#3304]
### Describe the build issues

YimMenu when I first got on the project was extremely lightweight, and only really depended on the logging framework, GTAV-Classes (tightly coupled), nlohmann JSON, ImGui and MinHook. All of those were tech debt from BigBase v2, but ignoring ImGui all of them were very lightweight and didn't affect the build that much.

Now with the introduction of an XML library, Lua, LibCURL, and now ASMJIT, the amount of tech debt the project has brought on is astronomical.

These projects need to be built as libraries instead of hard dependencies. Especially the Lua module, it is the bulkiest of all.

### IDE/Compiler

Visual Studio 2022

### Setup Guide

- [X] I confirm that I followed the setup guide correctly.

### YimMenu Source Code

- [X] I've confirmed this is an issue with building unmodified YimMenu.

**Comment by Rxann** [comment]
If the dependencies were not needed, trust me they would not be added. They allow YimMenu to expand and become a better menu. As Yimura said converting the feature to use MinHook and not ASMJIT would require extensive work that is really unneeded.

**Comment by lonelybud** [comment]
I agree on the fact that we should split out code into libraries for fast development process.
